### PR TITLE
added test for recreating deleted repos

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Adding test for recreating deleted Pulp Repositories

### CODE REVIEW

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -1312,6 +1313,49 @@ def test_module_stream_repository_crud_operations(self, repo):
...
+            org_id=module_org.id,
+            product=constants.PRDS['rhel'],
+            repo=constants.REPOS['rhst7']['name'],
+            reposet=constants.REPOSET['rhst7'],
+            releasever=None,
+        )
+        entities.Repository(id=repo_id).sync()
+        with default_sat.session.shell() as sh:
+            sh.send('foreman-rake console')
+            time.sleep(30)  # sleep to allow time for console to open
```
Comment (Griffin-Sullivan) [inline tests/foreman/api/test_repository.py:1344]: I tried testing to see if you could get any output out of `foreman-rake console`, but it seems to just return `None` as soon as the command is run.

**Comment by jyejare** [comment]
@ColeHiggins2 Please fix the `make docs` failing in the checks !

**Review by jyejare — APPROVED** [review]
ACK with non-blocking improvement comments !

File: tests/foreman/api/test_repository.py [hunk 2]
```diff
@@ -1312,6 +1313,49 @@ def test_module_stream_repository_crud_operations(self, repo):
         with pytest.raises(HTTPError):
             repo.read()
 
+    def test_positive_recreate_pulp_repositories(self, module_org, default_sat):
+        """Verify that deleted Pulp Repositories can be recreated using the
+        command 'foreman-rake katello:correct_repositories COMMIT=true'
+
+        :id: 2167d548-5af1-43e7-9f05-cc340d722aa8
+
+        :CaseImportance: High
```
[the remark this answers is not in the document]
Comment (jyejare) [reply tests/foreman/api/test_repository.py:1323]: I dont see its removed yet !

**Comment by jyejare** [comment]
@ColeHiggins2 Seems you have resolved PR comments except 1 for CaseImportance also please fix the checks failing in `make docs` by adding a tab for the failing line!

**Review by jyejare — APPROVED** [review]

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**